File: Exercises/Ex5/returned_answers/RichardFriedrichs/drawgraph.cxx
void drawgraph()
{
    TFile *file = TFile::Open("data.root");
    TTree *tree = (TTree *)file->Get("datatree");
    TH1F *hist = new TH1F("hist", "Data", 100, 5, 25);
    TCanvas *canvas = new TCanvas("canvas", "Histogram with Fit", 800, 600);
    tree->Draw("values>>hist");
    hist->SetFillColor(kYellow);
    hist->SetLineColor(kBlack);
    hist->SetLineWidth(3);
    hist->Draw();
    hist->GetXaxis()->SetTitle("Number");
    hist->GetYaxis()->SetTitle("Generated amount");
    hist->Fit("gaus");
    canvas->Update();
}

File: Exercises/Ex8/returned_answers/RichardFriedrichs/drawgraph.py
import ROOT
import numpy as np

ROOT.gROOT.SetBatch(True)

def drawgraph():

    values = []
    with open('masses.dat', 'r') as file:
        for line in file:
            line = line.strip()
            values.append(float(line))



    n_bins = len(values)
    hist = ROOT.TH1F("hist", "Mass distribution", n_bins, min(values), max(values))
    

    for mass in values:
        hist.Fill(mass)

    bw_function = ROOT.TF1("bw", "[0] * ( 1 / ((x**2-[1]**2)**2 + [1]**2*[2]**2 ))", min(values), max(values))
    bw_function.SetParameters(1.0, 125.0, 1.0)

    hist.Fit(bw_function, "R")
    k = bw_function.GetParameter(0)
    m = bw_function.GetParameter(1)
    gamma= bw_function.GetParameter(2)
    gamma = np.abs(gamma)
    

    canvas = ROOT.TCanvas("canvas", "Mass distribution", 800, 600)
    hist.Draw()
    #bw_function.Draw("SAME")
    hist.GetXaxis().SetTitle("Higgs mass in GeV")
    hist.GetYaxis().SetTitle("Number of events")
    canvas.Update()

    canvas.SaveAs("graph1.pdf")
    canvas.SaveAs("graph1.png")
    canvas.SaveAs("graph1.C")
    print("The width is",gamma)


if __name__ == "__main__":
    drawgraph()
